src/App.tsx

```diff
@@ -15,71 +16,99 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+// interface Good {
+//   name: string,
+//   id:number,
+// }
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+type State = {
+  selectedGood: string;
+};
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  selectGood = (good:string) => {
+    this.setState({ selectedGood: good });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  removeSelectedGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1
+          className={classNames('title', {
+            'is-flex': selectedGood,
+            'is-align-items-center': selectedGood,
+          })}
+        >
+          {!selectedGood ? (
+            'No goods selected'
+          ) : (
+            <>
+              {`${selectedGood} is selected`}
+              <button
+                aria-label="Select good"
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.removeSelectedGood}
+              />
+            </>
+          )}
+        </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const goodIsSelected = good === selectedGood;
```

Comment:
IMHO, if it is a bool value, you can start with "is"
```suggestion
              const isGoodSelected = good === selectedGood;
```